app/models/mixins/amazon_s3_authentication_mixin.rb

```diff
@@ -0,0 +1,86 @@
+module AmazonS3AuthenticationMixin
+  extend ActiveSupport::Concern
+
+  #
+  # Connections
+  #
+  def connect(options = {})
+    username = options[:username] || authentication_userid(options[:auth_type])
```

Comment:
These are defined in the `AuthenticationMixin` which happens to be included in `FileDepot`, but if any other class were to include `AmazonS3AuthenticationMixin` this would fail.

It feels like this file should be a part of the `FileDepotS3` class or maybe a utility in `/lib` for wrapping S3 connections in easier to use methods.

Reply:
So you'd prefer it if I pulled this out and included it in the FileDepotS3 class.  I can do that.

Reply:
Yeah, that makes sense (this having come from the provider repo). I think I would rather hide and simplify this for use just by the file depot then.